### PR TITLE
Model abstraction

### PR DESCRIPTION
**Pull request by kafoso**
@Sikian 

I've abstracted common functionality in models to other modules (or "classes"), allowing these model to extend parent models through the `inherit` module.

Most notably, the `User` model has been split up into `ThinkyDocumentModel` and `ConfigModel`; each extending each other.

This means:

1. It should be easier to adhere to the Open/closed principle.
2. Models are not forced to retrieve config data or the database Object on `init`, which in turn should improve performance.
3. It's easier to write manageable unit tests for each model's module file.

### CODE REVIEW

File: server/mvc/models/User.js [hunk 1]
```diff
@@ -1,109 +1,60 @@
...
-module.exports = function(){
-  var self = this,
-    document = new ThinkyModel({});
+var ThinkyModel;
+var getThinkyModel = function(thinky){
+  if (ThinkyModel === undefined) {
+    ThinkyModel = thinky.createModel("users", {
+      userId: thinky.type.number().integer().default(null).min(1).required().allowNull(false),
+      oauthType: thinky.type.string().default(null).min(1).required().allowNull(false),
+      oauthTokenEncrypted: thinky.type.string().default(null).min(1).required().allowNull(false),
```
Comment (Sikian) [inline server/mvc/models/User.js]: Maybe we should add the email to login with :)

**Comment by Sikian** [comment]
Really nice work here, I'd like to see it in action soon! :D

File: server/mvc/models/User.js (hunk 1 above)
Comment (kafoso) [inline server/mvc/models/User.js]: Yeah. I'll add that. I'm not done with the user flow, yet. All these annoying little things kept bubbling up. :)

**Comment by kafoso** [comment]
Thanks. It getting there... :)

File: server/mvc/models/User.js (hunk 1 above)
Comment (Sikian) [inline server/mvc/models/User.js]: Oh, I understand :)

**Pull request merged**